drivers/regex/mlx5/mlx5.c

```diff
@@ -16,13 +17,27 @@
 
 #include <rte_regexdev_driver.h>
 
+#include "mlx5.h"
+#include "mlx5_regex.h"
 #include "mlx5_regex_utils.h"
+#include "rxp-csrs.h"
 
 static TAILQ_HEAD(mlx5_regex_privs, mlx5_regex_priv) priv_list =
 					      TAILQ_HEAD_INITIALIZER(priv_list);
 static pthread_mutex_t priv_list_lock = PTHREAD_MUTEX_INITIALIZER;
 int mlx5_regex_logtype;
 
+struct mlx5_regex_db {
+	/* TODO: refer to struct rxp_database of Yuval code */
+	void *raw_mem;
+	struct mlx5dv_devx_umem *umem;
+	/* TODO: refer to Yuval's struct mlx5_database_ctx */
+	struct mlx5_database_ctx umem_ctx;
+};
+
+#define MLX5_REGEX_MAX_RXP_ENGINES 2
```

Comment:
This is no a define, it is a capability read from hca_cap.
So maybe need to initialize all relevant arrays after reading hca cap

Reply:
@yuvalavn ok, I will take fix as per the hca cap.